pilot/pkg/model/authentication_test.go

```diff
@@ -193,14 +193,6 @@ func TestGetPoliciesForWorkload(t *testing.T) {
 				constants.GatewayNameLabel: "my-gateway",
 			},
 			wantRequestAuthn: []*config.Config{
-				{
```

Comment:
Before, the policyattachment checker was doing 

```
if isGatewayAPI && len(targetRefs) == 0 && policy.GetSelector() != nil {
```

I believe that's what caused this test case to pass. We wouldn't hit the "ignore" for this branch because label selectors were nil. Now a nil label selector includes the policy. We can discuss how namespace-scoped policies should be handled.

Reply:
I think maybe we do want namespace (and mesh wide) policies to apply. @louiscryan ?